src/Common/Base32.cpp

```diff
@@ -0,0 +1,110 @@
+#include <Common/Base32.h>
+
+
+namespace DB
+{
+
+namespace
+{
+    Int8 getInitValue(UInt8 elem)
+    {
+        if (elem >= 'A' && elem <= 'Z')
+            return elem - 'A';
+        else if (elem >= '2' && elem <= '7')
+            return elem - '2' + 26;
+        return -1;
+    }
+}
+
+void encodeBase32(const UInt8 * src, size_t src_length, UInt8 * dst)
```

Comment:
I appreciate that you took the time to write your base32 implementation.

The thing is that ClickHouse is not in the base32-implementation business (it is also not in the business of writing unicode processing libraries, regexp libraries, etc.). In fact, base64 encoding/decoding (src/Functions/FunctionBase64Conversion.h) calls out into this [library](https://github.com/aklomp/base64). Yes, base58 is hand-written but I think this was a mistake. For base32, there is a [library](https://github.com/tplgy/cppcodec) with
- acceptable license (MIT),
- header-only build (if you like to integrate it into the Clickhouse build, look at contrib/annoy/ and contrib/annoy-cmake/ which is another header-only library),
- decent API as far as I see
- it supports all of the different base32 alphabet [craziness](https://en.wikipedia.org/wiki/Base32) - not that we would need it but it is nice-to-have anyways.

So my preference would be to proceed similar to base64, i.e. call out to a library instead.